packages/react-form-state/src/FormState.ts

```diff
@@ -91,13 +92,19 @@ export default class FormState<
     }
   }
 
-  state = createFormState(this.props.initialValues);
+  state = createFormState(this.props.initialValues, this.props.externalErrors);
   private mounted = false;
 
   componentDidMount() {
     this.mounted = true;
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
We shouldn't need this method. At the moment it's redundant with what we're doing in `createFormState`, and if we make the changes I suggested we won't need to alter the state based on `externalErrors` at all.

Reply:
From what I can tell, errors need to be associated with fields to stay consistent with how RemoteErrors are handled.
Several methods in the component relies on errors being associated with fields:
- `submit`: Prevent submitting of form before errors (remote or client error) are fixed
**How:** If `validateOnSubmit=true` then `hasClientErrors` checks if any field has error.
  - IMO, `hasFieldErrors` makes more sense, since remoteErrors are also associated with fields
- `updateField`: Easy of removing error in field by removing error on field change
**How:** Runs client validation on fields with errors (remote or client error).
In case the error is remote error, the client validation probably return null and error get removed.

In this case, we are associating external errors with fields following the pattern in the component.
We do not want to track externalErrors in state in order to conditionally derive new fields from external errors.

According to the React doc, `getDerivedStateFromProps` is not recommended for most components.
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

One of the suggested alternative is to use `componentDidUpdate`.
> If you need to perform a side effect (for example, data fetching or an animation) in response to a change in props, use componentDidUpdate lifecycle instead.

Reply:
We aren't performing a side-effect. There is no reason to use `componentDidUpdate` over `getDerivedStateFromProps` for mutating state.